.github/workflows/automatic-doc-checks.yml

```
@@ -0,0 +1,16 @@
+name: Main Documentation Checks
+
+on:
+  - push
+  - workflow_dispatch
```

I think we're close enough that we should just bite the bullet :)

```suggestion
  - pull_request
```